bcda/auth/api.go

```diff
@@ -26,28 +28,39 @@ import (
 		401: invalidCredentials
 		500: serverError
 */
+
 func GetAuthToken(w http.ResponseWriter, r *http.Request) {
 	clientId, secret, ok := r.BasicAuth()
 	if !ok {
 		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
 		return
 	}
 
-	token, expiresIn, err := GetProvider().MakeAccessToken(Credentials{ClientID: clientId, ClientSecret: secret})
+	tokenInfo, err := GetProvider().MakeAccessToken(Credentials{ClientID: clientId, ClientSecret: secret})
 	if err != nil {
-		http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
+		switch err.(type) {
+		case *customErrors.RequestTimeoutError:
+			//default retrySeconds: 1 second (may convert to environmental variable later)
+			retrySeconds := strconv.FormatInt(int64(1), 10)
+			w.Header().Set("Retry-After", retrySeconds)
+			http.Error(w, err.Error(), http.StatusServiceUnavailable)
+		case *customErrors.UnexpectedSSASError:
```

Comment:
Lets add *customErrors.InternalParsingError to also be here & write InternalServerError (not let it go to default 401 unauthorized)